src/content/questionnaire/sections/C.tsx

```diff
@@ -155,11 +178,24 @@ const FormSectionC: FC<FormSectionProps> = ({ SectionOption, refs }: FormSection
           name="species"
           options={speciesOptions.map((option) => ({ label: option, value: option }))}
           placeholder="Select species"
-          value={data.species}
+          value={species}
+          onChange={(val: string[]) => setSpecies(val)}
           multiple
           required
           readOnly={readOnlyInputs}
         />
+        <TextInput
+          id="section-c-other-species-of-subjects"
+          key={`other_species_involved_${otherSpecies?.toString()}`}
+          label="Other species involved"
+          name="otherSpeciesOfSubjects"
+          placeholder="Enter other species"
+          value={otherSpecies}
+          onChange={(e) => setOtherSpecies(e.target.value || "")}
+          maxLength={500}
+          required={species?.includes(CUSTOM_SPECIES.OTHER)}
+          readOnly={!species?.includes(CUSTOM_SPECIES.OTHER) || readOnlyInputs}
+        />
```

Comment:
Every keypress causes the "Other species involved" field to lose focus. Can this be fixed? The "Other cancer types" field doesn't seem to do it.

Reply:
Fixed the field losing focus, was just a typo in the key value. As for the other sections labels and titles you mentioned, I was told "... the only word we're worried about is "Subject", so the other labels you listed are fine as they are" (Referencing Species of subjects, Number of subjects labels)